pull-submodules.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -o nounset -o errexit
+
+MY_MODULES=$(git submodule status | awk '{print $2}')
+
+for m in $MY_MODULES; do
+    cd $m
+
+    echo $m
+
+    # Skip directories that have local modifications
+    git diff-index --quiet HEAD || { echo "  Skipping update of '$m' due to local modifications."; cd ..; continue; }
```

Comment:
`git status --porcelain`

Reply:
Not sure which one is better. `git diff-index --quiet HEAD` has the advantage (?) that untracked files do not prevent `git pull` from being run.

Reply:
Either way, as long as you're choosing purposefully.